src/Clapi/Server.hs

```diff
@@ -77,6 +77,9 @@ data ServerEvent ident a
     deriving (Eq, Show)
 type ServerEvent' = ServerEvent ClientAddr
 
+instance Show (Q.InChan a) where
+    show _ = "InChan"
```

Comment:
I have seen people use the Python convention of putting pointy brackets around values that are not directly representable: `<InChan>`. That would help, perhaps, given that the semantics of `show` originate around being able to serialise and deserialise (via `read`) what you'd write in Haskell itself.